### PR TITLE
Allow Users to Opt-out of Announcements

### LINKED ISSUE #8092: Allow users to opt-out of announcements
On the `/settings` page, we should provide an option for users with accounts to unsubscribe/turn off "announcement" broadcasts completely (similar to our checkbox for "display ads", which corresponds to `display_sponsors` on a `User`).

We could probably follow the convention and call the column on the table `display_announcements`.

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR gives Users the option to display or not display Announcements while browsing. This is accomplished by adding a `boolean` to the Users table, `display_announcements`, that defaults to `true` (similar to `display_sponsors`). On the backend, it also adds `display_announcements` to the permitted attributes in `user_policy.rb` and to `#user_data` in the `AsyncInfoController`. On the frontend, it adds an "Announcements" section to `/settings/misc`, where a User can toggle between displaying or not displaying Announcements.

## Related Tickets & Documents
Closes #8092 
Relates to #8090 and #8091 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
**Default (`display_announcements` is true):**
Screen Shot 2020-06-08 at 9 33 54 AM
**When a User does not want to display announcements:**
Screen Shot 2020-06-08 at 9 34 05 AM

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Announcement

### CODE REVIEW

**Comment by juliannatetreault** [comment]
> > Added tests? no, because they aren't needed until the Announcement work is merged in, making this feature testable!
> 
> I think we should definitely add some tests to make sure that a user can update these settings successfully, since the Rails updating aspect of this is in the scope of this PR!

I've added a spec to the `user_settings_spec.rb` that ensures that Announcements properly render under `/settings/misc` and I've added a new system spec (similar to the `user_selects_looking_for_work_spec.rb`),`user_updates_announcements_spec.rb`, that tests that Announcement settings can be updated. Let me know what you think!

File: spec/system/user/user_updates_announcements_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require "rails_helper"
+
+RSpec.describe "Announcements", type: :system do
```
Comment (vaidehijoshi) [inline spec/system/user/user_updates_announcements_spec.rb:3]:
```suggestion
RSpec.describe "User updates announcements", type: :system do
```

File: spec/system/user/user_updates_announcements_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  let(:user) { create(:user) }
+
+  before do
+    sign_in(user)
+    visit "/settings/misc"
+  end
+
+  it "renders the announcements settings" do
+    expect(page).to have_text("Announcements")
```
Comment (vaidehijoshi) [inline spec/system/user/user_updates_announcements_spec.rb:12]: What is this test actually checking for? It seems to do the same thing as the [one over here](https://github.com/thepracticaldev/dev.to/pull/8343/files#diff-dabdd41261904eae23b477c8ae3a61adR34-R37).

File: spec/system/user/user_updates_announcements_spec.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  it "properly updates the announcements settings", js: true do
+    page.check "Display Announcements (When browsing)"
+    sidekiq_perform_enqueued_jobs do
+      click_button("Save Announcements Settings")
+    end
+    expect(page).to have_text("Your profile was successfully updated.")
```
Comment (vaidehijoshi) [inline spec/system/user/user_updates_announcements_spec.rb:20]: Hm, this test checks that the flash message shows up, but if you think about it, it's not really testing that the broadcast was updated. For example, someone could go into the broadcasts controller, make a change to the `save!` method, and this flash message might still show up (meaning the test would pass even though the code didn't actually work!).

I think you want to take a look at `spec/requests/user/user_settings_spec.rb`... 👀

Comment (vaidehijoshi) [reply spec/system/user/user_updates_announcements_spec.rb:20]: I think we eventually _do_ want to add a system spec for announcements in order to test if the announcement banner does not appear if you uncheck the "display announcements" box (but that feature isn't hooked up yet 😆)

Currently though, I don't think that currently these tests are really testing _your_ changes to the backend.

**Pull request merged**